configs/components/_base-ruby-augeas.rb

```diff
@@ -35,11 +35,9 @@
 pkg.environment "CONFIGURE_ARGS", '--vendor'
 pkg.environment "PKG_CONFIG_PATH", "#{File.join(settings[:libdir], 'pkgconfig')}:/usr/lib/pkgconfig"
 
-if platform.is_solaris?
-  if platform.architecture == 'sparc'
-    pkg.environment "RUBY", host_ruby
-  end
+if platform.is_cross_compiled? && platform.is_solaris?
```

Comment:
This had the same issue as `ruby-shadow`, we shouldn't preload `rbconfig-xxx-orig.rb` for native compiles. And since `/opt/pl-build-tools/bin/gcc` is in the `PATH` (for Solaris 10) and `/opt/csw/bin/gcc` is in the `PATH` (for Solaris 11), then this component builds and installs successfully on `solaris-1*-i386` in `agent-runtime-7.x`:

```
2:14:58 export PATH="/usr/sbin:/usr/bin:/opt/pl-build-tools/bin:/usr/local/bin:/opt/csw/bin:/usr/ccs/bin:/usr/sfw/bin" && \
12:14:58 export CONFIGURE_ARGS="--vendor" && \
12:14:58 export PKG_CONFIG_PATH="/opt/puppetlabs/puppet/lib/pkgconfig:/usr/lib/pkgconfig" && \
12:14:58 gunzip -c "ruby-augeas-0.5.0.tgz" | /usr/sfw/bin/gtar xf - && \
12:14:58 : nothing to extract
12:14:58 touch ruby-augeas-unpack
12:14:58 touch ruby-augeas-patch
12:14:58 [ -d ruby-augeas-0.5.0 ] || mkdir -p ruby-augeas-0.5.0
12:14:58 touch ruby-augeas-configure
12:14:58 export PATH="/usr/sbin:/usr/bin:/opt/pl-build-tools/bin:/usr/local/bin:/opt/csw/bin:/usr/ccs/bin:/usr/sfw/bin" && \
12:14:58 export CONFIGURE_ARGS="--vendor" && \
12:14:58 export PKG_CONFIG_PATH="/opt/puppetlabs/puppet/lib/pkgconfig:/usr/lib/pkgconfig" && \
12:14:58 cd ruby-augeas-0.5.0 && \
12:14:58 /opt/puppetlabs/puppet/bin/ruby ext/augeas/extconf.rb && \
12:14:58 /opt/csw/bin/gmake -e -j3
12:14:59 creating Makefile
12:14:59 gmake[1]: Entering directory '/var/tmp/tmp.xRaaIe/ruby-augeas-0.5.0'
12:14:59 compiling ext/augeas/_augeas.c
12:14:59 linking shared-object _augeas.so
12:14:59 gmake[1]: Leaving directory '/var/tmp/tmp.xRaaIe/ruby-augeas-0.5.0'
12:14:59 touch ruby-augeas-build
12:14:59 touch ruby-augeas-check
12:14:59 export PATH="/usr/sbin:/usr/bin:/opt/pl-build-tools/bin:/usr/local/bin:/opt/csw/bin:/usr/ccs/bin:/usr/sfw/bin" && \
12:14:59 export CONFIGURE_ARGS="--vendor" && \
12:14:59 export PKG_CONFIG_PATH="/opt/puppetlabs/puppet/lib/pkgconfig:/usr/lib/pkgconfig" && \
12:14:59 cd ruby-augeas-0.5.0 && \
12:14:59 install -d '/opt/puppetlabs/puppet/lib/ruby/vendor_ruby' && \
12:14:59 cp -p 'lib/augeas.rb' '/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/augeas.rb' && \
12:14:59 chmod 0644 '/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/augeas.rb' && \
12:14:59 /opt/csw/bin/gmake -e -j3 DESTDIR=/ install && \
12:14:59 chown root:root /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/augeas.rb && \
12:14:59 /opt/csw/bin/gmake -e clean
12:14:59 gmake[1]: Entering directory '/var/tmp/tmp.xRaaIe/ruby-augeas-0.5.0'
12:14:59 /opt/csw/bin/ginstall -c -m 0755 _augeas.so //opt/puppetlabs/puppet/lib/ruby/vendor_ruby/2.7.0/i386-solaris2.10
12:14:59 gmake[1]: Leaving directory '/var/tmp/tmp.xRaaIe/ruby-augeas-0.5.0'
12:14:59 gmake[1]: Entering directory '/var/tmp/tmp.xRaaIe/ruby-augeas-0.5.0'
12:14:59 gmake[1]: Leaving directory '/var/tmp/tmp.xRaaIe/ruby-augeas-0.5.0'
```